src/runtime/services/HttpService.ts

```
@@ -57,8 +57,10 @@ export default class HttpService {
               Object.assign(context, ctx);
             },
           });
-        } catch {
-          await this.onAuthFailure(AuthStatus.Expired);
+        } catch (e) {
+          if ((e as (Partial<FetchError>)).status === HTTP_STATUS_UNAUTHORIZED) {
+            await this.onAuthFailure(AuthStatus.Expired);
+          }
```

Please throw the error in the else case.